### PR TITLE
Refactor visual settings

### PR DESCRIPTION
**Pull request by UselessToucan**
Reduce code duplication related to getting bindables from config

### CODE REVIEW

File: osu.Game/OsuGameBase.cs [hunk 1]
```diff
@@ -110,6 +110,10 @@ private void load()
             dependencies.CacheAs(this);
             dependencies.Cache(LocalConfig);
 
+            BindableVisualSettings visualSettings;
+            AddInternal(visualSettings = new BindableVisualSettings());
+            dependencies.Cache(visualSettings);
```
Comment (smoogipoo) [inline osu.Game/OsuGameBase.cs]: I really don't like visual settings being global like this, and especially not at an OsuGameBase level.

File: osu.Game/Configuration/BindableVisualSettings.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+
+namespace osu.Game.Configuration
+{
+    public class BindableVisualSettings : Component
```
Comment (smoogipoo) [inline osu.Game/Configuration/BindableVisualSettings.cs]: It's not a bindable, so it shouldn't be called a bindable.

File: osu.Game/Skinning/LocalSkinOverrideContainer.cs [hunk 3]
```diff
@@ -82,13 +82,13 @@ protected override IReadOnlyDependencyContainer CreateLocalDependencies(IReadOnl
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuConfigManager config)
+        private void load(BindableVisualSettings visualSettings)
         {
-            beatmapSkins = config.GetBindable<bool>(OsuSetting.BeatmapSkins);
+            beatmapSkins = visualSettings.BeatmapSkins;
```
Comment (smoogipoo) [inline osu.Game/Skinning/LocalSkinOverrideContainer.cs]: In general, we don't ever want to keep non-weak-reference upwards references like how it's done here and in the other classes that use visualSettings this way.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
I don't see the point of this PR in general. This doesn't simplify anything.

@peppy requesting second opinion.

**Comment by peppy** [comment]
Going forward we will definitely need more compartmentalised config stores, but still will come once we move game level config to be database-backed. I don't think it's worthwhile to make a change like this right now, even if it was put in the correct place (probably `PlayerLoader`) and the bindable issues were fixed.

**Pull request closed without merging**